app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -18,9 +17,8 @@ class AgendaRepositoryImpl(
         return try {
             val response = api.getAgendaItems(timezone = timezone, time = time)
             val tasks = response.taskDtos.map { task ->
-                AgendaItem(
-                    apiId = task.id,
-                    type = AgendaItemType.TASK,
+                AgendaItem.Task(
+                    id = task.id.toInt(),
```

Comment:
Why are these int IDs? You don't need separate IDs for local DB and API, you can just use the unique API IDs as primary key